### PR TITLE
Fix flaky crypto playwright tests

### PR DESCRIPTION
**Pull request by richvdh**
See the commit comments for details.

Fixes https://github.com/element-hq/element-web/issues/28061

### CODE REVIEW

File: playwright/pages/client.ts [hunk 1]
```diff
@@ -289,6 +289,42 @@ export class Client {
...
+    public async awaitRoomMembership(roomId: string, membership: string = "join") {
+        await this.evaluate(
+            (cli: MatrixClient, { roomId, membership }) => {
+                const isReady = () => {
+                    // Fetch the room on each check, because we get a different instance before and after the join arrives.
+                    const room = cli.getRoom(roomId);
+                    const myMembership = room?.getMyMembership();
+                    // @ts-ignore access to private field "logger"
+                    cli.logger.info(`waiting for room ${roomId}: membership now ${myMembership}`);
+                    return myMembership == membership;
```
Comment (MidhunSureshR) [inline playwright/pages/client.ts:309]:
```suggestion
                    return myMembership === membership;
```

File: playwright/pages/client.ts [hunk 2]
```diff
@@ -289,6 +289,42 @@ export class Client {
...
+                };
+                if (isReady()) return;
+
+                return new Promise<void>((resolve) => {
```
Comment (MidhunSureshR) [inline playwright/pages/client.ts:313]: could be useful to reject  this promise after a timeout with a sensible error.

**Review by MidhunSureshR — APPROVED** [review]
Not the assigned reviewer but this PR is useful for merging https://github.com/element-hq/matrix-react-sdk/pull/84/

File: playwright/pages/client.ts (hunk 2 above)
Comment (richvdh) [reply playwright/pages/client.ts]: Agreed. done.

**Pull request merged**